test_conformance/vulkan/vulkan_interop_common/vulkan_wrapper.cpp

```diff
@@ -0,0 +1,2228 @@
+
+VulkanPipelineLayout::VulkanPipelineLayout(
+    const VulkanPipelineLayout & pipelineLayout)
+    : m_device(pipelineLayout.m_device),
+      m_vkPipelineLayout(pipelineLayout.m_vkPipelineLayout)
+{
+
+}
+
+void
+VulkanPipelineLayout::VulkanPipelineLayoutCommon(
+    const VulkanDescriptorSetLayoutList & descriptorSetLayoutList)
+{
+    VkPipelineLayoutCreateInfo vkPipelineLayoutCreateInfo = {};
+    vkPipelineLayoutCreateInfo.sType                  = VK_STRUCTURE_TYPE_PIPELINE_LAYOUT_CREATE_INFO;
+    vkPipelineLayoutCreateInfo.pNext                  = NULL;
+    vkPipelineLayoutCreateInfo.flags                  = 0;
+    vkPipelineLayoutCreateInfo.setLayoutCount         = (uint32_t)descriptorSetLayoutList.size();
+    vkPipelineLayoutCreateInfo.pSetLayouts            = descriptorSetLayoutList;
+    vkPipelineLayoutCreateInfo.pushConstantRangeCount = 0;
+    vkPipelineLayoutCreateInfo.pPushConstantRanges    = NULL;
+
+    vkCreatePipelineLayout(m_device, &vkPipelineLayoutCreateInfo, NULL, &m_vkPipelineLayout);
+}
+
+VulkanPipelineLayout::VulkanPipelineLayout(
+    const VulkanDevice & device,
+    const VulkanDescriptorSetLayout & descriptorSetLayout)
+    : m_device(device),
+      m_vkPipelineLayout(VK_NULL_HANDLE)
+{
+    VulkanDescriptorSetLayoutList descriptorSetLayoutList;
+    descriptorSetLayoutList.add(descriptorSetLayout);
+
+    VulkanPipelineLayoutCommon(descriptorSetLayoutList);
+}
+
+VulkanPipelineLayout::VulkanPipelineLayout(
+    const VulkanDevice & device,
+    const VulkanDescriptorSetLayoutList & descriptorSetLayoutList)
+    : m_device(device),
+      m_vkPipelineLayout(VK_NULL_HANDLE)
+{
+    VulkanPipelineLayoutCommon(descriptorSetLayoutList);
+}
+
+VulkanPipelineLayout::~VulkanPipelineLayout()
+{
+    vkDestroyPipelineLayout(m_device, m_vkPipelineLayout, NULL);
+}
+
+VulkanPipelineLayout::operator VkPipelineLayout() const
+{
+    return m_vkPipelineLayout;
+}
+
+///////////////////////////////////////
+// VulkanShaderModule implementation //
+///////////////////////////////////////
+
+VulkanShaderModule::VulkanShaderModule(
+    const VulkanShaderModule & shaderModule)
+    : m_device(shaderModule.m_device),
+      m_vkShaderModule(shaderModule.m_vkShaderModule)
+{
+
+}
+
+VulkanShaderModule::VulkanShaderModule(
+    const VulkanDevice & device,
+    const std::string & code)
+    : m_device(device)
+{
+    std::string paddedCode = code;
+    while (paddedCode.size() % 4) {
+        paddedCode += " ";
+    }
+
+    VkShaderModuleCreateInfo vkShaderModuleCreateInfo = {};
+    vkShaderModuleCreateInfo.sType    = VK_STRUCTURE_TYPE_SHADER_MODULE_CREATE_INFO;
+    vkShaderModuleCreateInfo.pNext    = NULL;
+    vkShaderModuleCreateInfo.flags    = 0;
+    vkShaderModuleCreateInfo.codeSize = paddedCode.size();
+    vkShaderModuleCreateInfo.pCode    = (const uint32_t *)(void *)paddedCode.c_str();
+
+    vkCreateShaderModule(m_device, &vkShaderModuleCreateInfo, NULL, &m_vkShaderModule);
```

Comment:
Creating a shader module from GLSL code is not standard Vulkan. We either need to pre-compile the SPIR-V or integrate glslang.

Reply:
Ack. We are working on updating this to avoid using NV extension.